src/tests/key-unlock.cpp

```diff
@@ -147,7 +147,11 @@ TEST_F(rnp_tests, test_key_unlock_pgp)
                       std::ios_base::binary | std::ios_base::out | std::ios_base::in);
     off_t        versize = file_size("dummyfile.dat.pgp");
     verf.seekg(versize - 3, std::ios::beg);
-    verf.write("\x0C", 1);
+    char bt = 0;
+    verf.read(&bt, 1);
+    bt = bt ^ 0xff;
+    verf.seekg(versize - 3, std::ios::beg);
```

Comment:
Can you use peek instead of seeking twice?

Reply:
Good catch by the way.

Reply:
@dewyatt Sure. Will force-push in a moment.

Reply:
..and thanks! )

Reply:
@dewyatt Hm, fstream appears to work in a bit strange way, as from https://stackoverflow.com/questions/15670359/fstream-seekg-seekp-and-write

> The implications of point 2 are that between reads and writes on an fstream you must either flush the buffer or seek the file position when changing from output to input, and you must either be at the end of the file or seek the file position when changing from input to output.`

Anyway, changed code a bit to not do call to `file_size()`

Reply:
@ni4 The wording is confusing there, but I believe the second seek call is now redundant because we're not reading or writing at that point, we're peeking.

However, it's just one line, I'm not too worried about it.